client/components/data/query-domain-owner-username/index.ts

```diff
@@ -0,0 +1,41 @@
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	login: string;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useDomainOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domain: ResponseDomain | null | undefined
+) {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const domainSubscription = purchases.filter(
+		( purchase ) => purchase.id === parseInt( domain?.subscriptionId ?? '0' )
+	)[ 0 ];
```

Comment:
Not a big deal but it seems a bit strange to evaluate the identifier of the domain subscription in that loop, maybe we should extract that:

```suggestion
	const domainSubscriptionId = domain?.subscriptionId ?? '0';

	const domainSubscription = purchases.filter(
		( purchase ) => purchase.id === parseInt( domainSubscriptionId )
	)[ 0 ];
```